### ISSUE TITLE
Runtime error when trying to resolve dependencies [after 1.0.0 upgrade]

### ISSUE THREAD

**Issue by ltcmelo** [daveaglick/Buildalyzer#65]
Hey @daveaglick , I just gave it a try with the just released 1.0.0 version. I upgraded my Roslyn versions and refactored the API accordingly. Everything builds fine, but when I try to analyse my test projects I get an exception out of `ProjectAnalyzer.Build`.

From the trace below, I suspect this could be to a messed up Windows/Unix path separator (i.e., `/` versus `\`)?

```
/usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(198,5): error MSB4018: The "ResolvePackageAssets" task failed unexpectedly.
      /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(198,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'NuGet.Frameworks, Version=4.7.0.5, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the path specified.
      /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(198,5): error MSB4018:
      /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(198,5): error MSB4018: File name: 'NuGet.Frameworks, Version=4.7.0.5, Culture=neutral, PublicKeyToken=31bf3856ad364e35' ---> System.IO.DirectoryNotFoundException: Could not find a part of the path '/usr/local/share/dotnet/sdk/2.1.302/Sdks/..\'.
      /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(198,5): error MSB4018:    at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path)
```

**Comment by daveaglick** [comment]
Very well could be an issue with non-Windows platforms. I'm currently trying to get my CI build to work (fighting some annoying differences between the VS test runner and `dotnet test` at the moment). Once I've got that up and running, I'll be able to push a patch to MyGet that I'd like you to try.

**Comment by ltcmelo** [comment]
I created a PR to fix this error (and a related one): https://github.com/daveaglick/Buildalyzer/pull/67/files

I've also identified places in `EnvironmentFactory` where directory separators are being used on a Windows-specific manner. But I'll create separate patches for those.

**Comment by daveaglick** [comment]
Awesome, thanks! I'm in the process of getting CI working on both Windows and Ubuntu via AppVeyor (as you can see from my recent commits). I'll get this merged right after - it'll be a good indication that my Linux builds are working correctly. Has this change been tested locally on your non-Windows system?

**Comment by ltcmelo** [comment]
We do run our tool on Windows, for .NET Framework applications. But we're not yet complete enough in terms of *testing*. So I'd recommend that you test it yourself - feel free to cherry pick from the branch (I'll keep that branch evolving, but I'm not sure whether general changes might interest you).

**Comment by ltcmelo** [comment]
Oh, misread your question: in a non-Windows system, yes, that fixes the issue for me (I'm on OSX).

**Comment by ltcmelo** [comment]
One more comment... You'll notice that I also made path separators cross-platform even for .NET Framework (i.e., on Windows) stuff. This is because our tool allows static analysis of .NET Framework applications on non-Windows system, by requiring that the user copies VS Build tools and reference assemblies files and places them in a way that their directory organization mimics the original one in Windows.

**Comment by ltcmelo** [comment]
Done: https://github.com/daveaglick/Buildalyzer/pull/68

If you're OK with, please give permission to push branches to your repository, this would make things easier for me when submitting PRs (otherwise, I need to push them on "my upstream" first).

**Comment by ltcmelo** [comment]
Hum... Think I did a little mess during merge. Hold on.

**Comment by ltcmelo** [comment]
Fixed. :)

**Comment by daveaglick** [comment]
I just published Buildalyzer 2.0 which runs MSBuild out-of-process (as opposed to the old way of running it via MSBuild APIs). In general, if you can build it Buildalyzer should now be able to build it. Given how drastically different this new technique is, I'm closing all existing bug reports.

Please try the new version and if you're still having problems, please open a new issue.
